packages/core/android/src/main/java/io/sentry/react/RNSentryReactFragmentLifecycleTracer.java

```diff
@@ -95,6 +107,93 @@ public void onEventDispatch(Event event) {
             }
           }
         });
+
+    // Add layout listener to detect configuration changes
+    attachLayoutChangeListener(v);
+  }
+
+  @Override
+  public void onFragmentViewDestroyed(@NonNull FragmentManager fm, @NonNull Fragment f) {
+    detachLayoutChangeListener();
+  }
+
+  private void attachLayoutChangeListener(final View view) {
+    final ViewTreeObserver.OnGlobalLayoutListener listener =
+        new ViewTreeObserver.OnGlobalLayoutListener() {
+          @Override
+          public void onGlobalLayout() {
+            checkAndNotifyWindowSizeChange(view);
+          }
+        };
+
+    currentView = view;
+    currentListener = listener;
+
+    view.getViewTreeObserver().addOnGlobalLayoutListener(listener);
+  }
+
+  private void detachLayoutChangeListener() {
+    if (currentView != null && currentListener != null) {
+      try {
+        currentView.getViewTreeObserver().removeOnGlobalLayoutListener(currentListener);
+      } catch (Exception e) {
+        logger.log(SentryLevel.DEBUG, "Failed to remove layout change listener", e);
+      }
+    }
+
+    currentView = null;
+    currentListener = null;
+  }
+
+  private void checkAndNotifyWindowSizeChange(View view) {
+    try {
+      android.util.DisplayMetrics metrics = view.getContext().getResources().getDisplayMetrics();
+      int currentWidth = metrics.widthPixels;
+      int currentHeight = metrics.heightPixels;
+
+      if (lastWidth != currentWidth || lastHeight != currentHeight) {
+        lastWidth = currentWidth;
+        lastHeight = currentHeight;
+
+        notifyReplayIntegrationOfSizeChange(currentWidth, currentHeight);
+      }
+    } catch (Exception e) {
+      logger.log(SentryLevel.DEBUG, "Failed to check window size", e);
+    }
+  }
+
+  private void notifyReplayIntegrationOfSizeChange(int width, int height) {
+    try {
```

Comment:
If you accept https://github.com/getsentry/sentry-react-native/pull/5321/files#r2481124409 we will need to remove this code and format the below code